src/orquestra/sdk/_base/_storage/orqdantic.py

```diff
@@ -0,0 +1,123 @@
+################################################################################
+# © Copyright 2024 Zapata Computing Inc.
+################################################################################
+
+"""Compatibility layer for pydantic v1 / v2 compatibility."""
+
+from copy import deepcopy
+from typing import Annotated, Any, Optional
+
+import pydantic
+
+PYDANTICV1 = pydantic.__version__.startswith("1.")
+
+if PYDANTICV1:
```

Comment:
we can hack around the typechecking with:

```suggestion
from typing import TYPE_CHECKING


if PYDANTICV1 and not TYPE_CHECKING:
```